src/parser/xasl_generation.c

```diff
@@ -21596,7 +21596,8 @@ pt_to_update_xasl (PARSER_CONTEXT * parser, PT_NODE * statement, PT_NODE ** non_
     }
 
 cleanup:
-  if (aptr_statement != NULL)
+  /* postphoned freeing for CTE query execution */
+  if (aptr_statement != NULL && !pt_is_allowed_result_cache ())
```

Comment:
If `aptr_statement != NULL` and `(!pt_is_allowed_result_cache ()) == false`, where do you free aptr_statement?

Reply:
maybe, it's free at the end of compiling.